aten/src/ATen/VmapTransforms.cpp

```diff
@@ -80,6 +80,17 @@ VmapDimVector VmapPhysicalView::getPhysicalShape(IntArrayRef logical_shape) cons
   return result;
 }
 
+VmapDimVector VmapPhysicalView::getPhysicalStrides(IntArrayRef logical_strides) const {
+  at::VmapDimVector physical_strides;
+  auto batch_strides = tensor().strides().slice(0, numBatchDims());
+  physical_strides.reserve(numBatchDims() + logical_strides.size());
+  physical_strides.insert(
+      physical_strides.end(), batch_strides.begin(), batch_strides.end());
+  physical_strides.insert(
+      physical_strides.end(), logical_strides.begin(), logical_strides.end());
+  return physical_strides;
+}
+
```

Comment:
This should not be a helper function. It's possible that the strides of the batch dims don't "match" the logical strides. Like if you have a tensor of size(2, 3, 4) and all dimensions are batch dims, and one passes in logical_strides(5, 1), then we're going to end up creating a tensor with strides (12, 4, 1, 5, 1) which is bogus: what the user might have wanted is a tensor with stride (2 * 3 * 4 * 5, 3 * 4 * 5, 4 * 5, 5, 1)